test/powershell/Modules/Microsoft.PowerShell.Security/CmsMessage.Tests.ps1

```diff
@@ -0,0 +1,400 @@
+        $recipient.Resolve($executionContext.SessionState, "Decryption", [ref] $ers)
+
+        $recipient.Certificates.Count | Should Be 1
+    }
+
+    It "Verify resolution by thumbprint" {
+        $ers = $null
+        $recipient = [System.Management.Automation.CmsMessageRecipient] $importedCert.Thumbprint
+        $recipient.Resolve($ExecutionContext.SessionState, "Decryption", [ref] $ers)
+
+        # "Should have certs from thumbprint in 'My' store"
+        $recipient.Certificates[0].Thumbprint | Should Be $importedCert.Thumbprint
+    }
+
+    It "Verify resolution by subject name" {
+        $ers = $null
+        $recipient = [System.Management.Automation.CmsMessageRecipient] $importedCert.Subject
+        $recipient.Resolve($ExecutionContext.SessionState, "Decryption", [ref] $ers)
+
+        $recipient.Certificates[0].Thumbprint | Should Be $importedCert.Thumbprint
+    }
+
+    It "Verify error when no cert found in encryption for encryption" {
+        $ers = $null
+        $recipient = [System.Management.Automation.CmsMessageRecipient] "SomeCertificateThatDoesNotExist*"
+        $recipient.Resolve($ExecutionContext.SessionState, "Encryption", [ref] $ers)
+
+        $ers[0].FullyQualifiedErrorId | Should Be "NoCertificateFound"
+    }
+
+    It "Verify error when encrypting to non-wildcarded identifier for decryption" {
+        $ers = $null
+        $recipient = [System.Management.Automation.CmsMessageRecipient] "SomeCertificateThatDoesNotExist"
+        $recipient.Resolve($ExecutionContext.SessionState, "Decryption", [ref] $ers)
+
+        $ers[0].FullyQualifiedErrorId | Should Be "NoCertificateFound"
+    }
+
+    It "Verify error when encrypting to wrong cert" {
+        $ers = $null
+        $goodCerts = @(Get-ChildItem Cert:\currentuser\My -DocumentEncryptionCert | ForEach-Object Thumbprint)
+        $badCert = Get-ChildItem Cert:\currentuser\My | ? { $_.Thumbprint -notin $goodCerts } | Select -First 1
+
+        $recipient = [System.Management.Automation.CmsMessageRecipient] $badCert.Thumbprint
+        $recipient.Resolve($ExecutionContext.SessionState, "Encryption", [ref] $ers)
+
+        $ers[0].FullyQualifiedErrorId | Should Be "CertificateCannotBeUsedForEncryption"
+    }
+
+    It "Verify no error when encrypting to wildcarded identifier for decryption" {
+        $ers = $null
+        $recipient = [System.Management.Automation.CmsMessageRecipient] "SomeCertificateThatDoesNotExist*"
+        $recipient.Resolve($ExecutionContext.SessionState, "Decryption", [ref] $ers)
+
+        $ers | Should Be $null
+        $recipient.Certificates.Count | Should Be 0
+    }
+
+    It "Verify Protect-CmsMessage emits recipient errors" {
+        try {
+            "Hello World" | Protect-CmsMessage -To "SomeThumbprintThatDoesNotExist" -ErrorAction Stop
+            throw "No Exception!"
+        } catch {
+            $_.FullyQualifiedErrorId | Should Be "NoCertificateFound,Microsoft.PowerShell.Commands.ProtectCmsMessageCommand"
```

Comment:
I put the same comment in that PR. I will postpone changing to this pattern for now.